hasura.planx.uk/migrations/1731498724636_run_sql_migration/up.sql

```diff
@@ -0,0 +1,160 @@
+    ls.created_at,
+    ls.submitted_at,
+    ((ls.submitted_at)::date - (ls.created_at)::date) AS session_length_days,
+    ls.has_user_saved AS user_clicked_save,
+    rps.number_times_resumed,
+    ls.allow_list_answers,
+    ((ls.allow_list_answers -> 'proposal.projectType'::text))::text AS proposal_project_type,
+    ((ls.allow_list_answers -> 'application.declaration.connection'::text))::text AS application_declaration_connection,
+    ((ls.allow_list_answers -> 'property.type'::text))::text AS property_type,
+    ((ls.allow_list_answers -> 'drawBoundary.action'::text))::text AS draw_boundary_action,
+    ((ls.allow_list_answers -> 'user.role'::text))::text AS user_role,
+    ((ls.allow_list_answers -> 'property.constraints.planning'::text))::text AS property_constraints_planning,
+        CASE
+            WHEN (((pr.payment_requests)::jsonb IS NOT NULL) AND (jsonb_array_length((pr.payment_requests)::jsonb) > 0)) THEN true
+            ELSE false
+        END AS user_invited_to_pay,
+    pr.payment_requests,
+    ps.payment_status,
+        CASE
+            WHEN (((ba.bops_applications)::jsonb IS NOT NULL) AND (jsonb_array_length((ba.bops_applications)::jsonb) > 0)) THEN true
+            ELSE false
+        END AS sent_to_bops,
+    ba.bops_applications,
+        CASE
+            WHEN (((ua.uniform_applications)::jsonb IS NOT NULL) AND (jsonb_array_length((ua.uniform_applications)::jsonb) > 0)) THEN true
+            ELSE false
+        END AS sent_to_uniform,
+    ua.uniform_applications,
+        CASE
+            WHEN (((ea.email_applications)::jsonb IS NOT NULL) AND (jsonb_array_length((ea.email_applications)::jsonb) > 0)) THEN true
+            ELSE false
+        END AS sent_to_email,
+    ea.email_applications,
+    ((ls.allow_list_answers -> 'findProperty.action'::text))::text AS find_property_action,
+        CASE
+            WHEN (((sa.s3_applications)::jsonb IS NOT NULL) AND (jsonb_array_length((sa.s3_applications)::jsonb) > 0)) THEN true
+            ELSE false
+        END AS sent_to_s3_power_automate,
+    sa.s3_applications,
+    ((ls.allow_list_answers -> 'usedFOIYNPP'::text))::text AS used_foiynpp,
+    ((ls.allow_list_answers -> 'propertyInformation.action'::text))::text AS property_information_action,
+    ((ls.allow_list_answers -> 'planningConstraints.action'::text))::text AS planning_constraints_action,
+    ((ls.allow_list_answers -> '_overrides'::text))::text AS overrides,
+    ((ls.allow_list_answers -> 'rab.exitReason'::text))::text AS rab_exit_reason,
+    ((ls.allow_list_answers -> 'service.type'::text))::text AS pre_app_service_type,
+    ((ls.allow_list_answers -> 'application.information.harmful'::text))::text AS pre_app_harmful_info,
+    ((ls.allow_list_answers -> 'application.information.sensitive'::text))::text AS pre_app_sensitive_info,
+    (((ls.allow_list_answers -> 'application.type'::text) -> 0))::text AS application_type,
+    ((ls.allow_list_answers -> '_feedback'::text))::text AS feedback
```

Comment:
Question: since `_feedback` is an obejct itself, should the views actually drill down one level further to, for example, expose feedback score as its own column? Otherwise I think we'll hit up against usual "Metabase can't process JSONB types" error ! 

Score seems like obvious first priority for analytics, I'd leave it up to you if we want to expose other `_feedback` fields too now/up front or wait to expose them on request later if it comes up!

Reply:
Have attempted to implement this, but wasn't really very sure of it!

Also struggled to test it - eventually realised it was only populating the analytics tables when I finished a flow - is that right? 😅

Reply:
Full comment below - let's try something like `((ls.allow_list_answers -> '_feedback' -> 'feedbackScore'::int))::int AS feedback_score` to extract the view-level column data !